### ISSUE TITLE
[HELP]: Can we use this plugin which has LGPL 3.0 license for our startup?

### ISSUE THREAD

**Issue by SatyamKr07** [Canardoux/flutter_sound#696]
## Here is my question :

I have a startup idea and want to use this plugin. Our app code won't be public as it is going to be a startup. So can I still use it?

**Comment by cedvdb** [comment]
This is a license question. Unrelated to this repository.

**Comment by Larpoux** [comment]
Flutter Sound is LGPL 3.0
This means that you can freely use the library inside a proprietary App.
But if you want to fork and develop your own version of Flutter Sound, you will have to publish your modifications under a LGPL license, too.
Also, if you include part or all the **source code** inside a project, you will have to publish your App under the LGPL license, too.

**Comment by BjoernPetersen** [comment]
A LGPL library may only be used in a proprietary app if the library is dynamically linked. Dart links all dependencies statically and includes them in one binary, so the license effectively prohibits usage in a closed source project.

**Comment by Larpoux** [comment]
I did not know that LGPL software must be dynamicaly linked with the App.
If you are right, this is a big problem.
I would not like to have to change the Flutter Sound license.
But many Flutter Sound developers use Flutter Sound in a proprietary App.

I am going to check that.

**Comment by Larpoux** [comment]
@BjoernPetersen ,
Yes you are right : the standard LGPL license does not allow static linked libraries.
I am surprised that nobody never say anything about that.
Probably the solution would be : 
- Either to add an exception in the Flutter Sound license to allow static links (some product like xwWidgets have such exceptions)
- Or change the license to Mozilla (MGPL).


We really want that the developers can use Flutter Sound in a proprietary close source App. But changing the license would be time consuming : I have many things to do to improve Flutter Sound. If I spend time on license issues, it will be instead of other useful enhancements. But still I agree with you license issues are something important

**Comment by Larpoux** [comment]
OK, I changed the License of the next Flutter Sound release to be MPL2.0.
This change will be for Flutter Sound 9.0.x-beta (or maybe 8.4.x).

Is this OK for you ?

**Comment by BjoernPetersen** [comment]
I'd love it if this library was licensed under the MPL2.0, but you have to be careful. Looking at the GitHub stats, about forty people contributed to this project, and they did that under the LGPL license. I'm not a lawyer or anything, but as far as I understand you can only change the license for code you actually wrote, and you'd need to get the consent of everyone else to relicense their code as well.

**Comment by Larpoux** [comment]
If I understand everything, many developers are using Flutter Sound for proprietary/close source App but are not compliant with LGPL because LGPL does not allow static links.
I really do not like working for private software. But this is actually like that. So it is urgent to change the License.

If someone sue me and send me to jail, I will have much time to work on Free and Open Source software 😆 .
But there is a problem : I will not have a computer and there is no internet in jails. :/

**Comment by Milluz** [comment]
Hello @Larpoux, 
I'm interested in use flutter_sound on my app but the lgpl license blocks me too. I see that you are changing it to MPL, and here is my question: You wrote about the tau react license is GPL and there's a tao core submodule on the source, so if you are using it inside flutter_sound even if you change the flutter_sound to MPL the GPL is strong right? Sorry if I missing something here.

**Comment by Larpoux** [comment]
Hi @Milluz ,

You are right : GPL is a very strong license.
Actually there is absolutely **no code** covered by a GPL license. τ React, for example, is just a dream in my head.

But @BjoernPetersen noticed that the LGPL license was not correct, because Flutter is statically linked with the App.

My plan is to fork flutter_sound  8.3 to two different branches : 
- flutter_sound 8.4, covered by a MPL2 license
- tau_sound 9.0, which will be protected by a pure GPL license, (because I do not want to work on a product not correctly published with a correct GPL license).

τ Sound 9.0 will hopefully have many new great things (I hope) : 
- Better Web Support
- A port to React Native
- A port to Microsoft Blazor
- A new class which will implement Audio Graphs 
- Better PCM support
- ... and perhaps other fantastic things 😉 

There is actually no problem for you, to use the current Flutter Sound plugin inside an App which can have the License that you decide. (Many Flutter Sound users use Flutter Sound inside proprietary close source Apps).

If I manage to do the fork that I consider, you will decide if you want to use Flutter Sound V8.4 (MPL) or Tau Sound V9.0 (GPL)

**Comment by ryanheise** [comment]
> I'd love it if this library was licensed under the MPL2.0, but you have to be careful. Looking at the GitHub stats, about forty people contributed to this project, and they did that under the LGPL license. I'm not a lawyer or anything, but as far as I understand you can only change the license for code you actually wrote, and you'd need to get the consent of everyone else to relicense their code as well.

IANAL either, but yes this is also my understanding. It is not just a matter of whether it puts the maintainer at risk of being sued for re-licensing other people's code without their consent, but it is also that companies put trust into your copyright notice and license that you do indeed have the right to license all of that code under the new license. If that's not the case, then you may be causing some difficult legal situations for the people and organisations who use this plugin.

How about trying to contact all of the contributors and ask them if they consent? It's not as tricky as the ffmpeg situation where there are hundreds of contributors and it would be practically impossible to not only reach them all but also convince them all that your alternative license is better than the one they originally agreed to use. Here, there are only around 40-50 people to contact.

To make that easier, you can run this command to get a list of emails, and then send out a mass email using the BCC header to avoid every recipient then bothering each other:

```sh
git log|grep Author|sed 's/^Author: //g'|sort|uniq
```

If most consent but a few don't consent, then you may be able to remove just their bits of code and rebuild those bits with non-infringing code. However, if you explain the actual issue with the (L)GPL license and why the dynamic linking clause is actually inncompatible with the Apple and Google mobile platforms, there's a good chance they'll agree to support the move.
